pkg/sbom/cyclonedx/marshal.go

```diff
@@ -315,11 +321,17 @@ func pkgComponent(pkg Package) (*core.Component, error) {
 	}
 
 	name := pkg.Name
+	version := pkg.Version
 	var group string
-	// use `group` field for GroupID and `name` for ArtifactID for jar files
-	if pkg.Type == ftypes.Jar {
-		name = pu.Name
-		group = pu.Namespace
+	// there are cases when we can't build purl
+	// e.g. local Go packages
+	if pu != nil {
+		version = pu.Version
```

Comment:
I'm probably missing something. Why do we need to override the version with the PURL version here?

Reply:
We are currently using `PURL` version -https://github.com/aquasecurity/trivy/blob/e2fb3dd58f2800ffbeb106ace8e30032c2600e1e/pkg/sbom/cyclonedx/marshal.go#L372

I think it has to do with the version format.
We merge `version`, `release` and `epoch` for PURL - https://github.com/aquasecurity/trivy/blob/e2fb3dd58f2800ffbeb106ace8e30032c2600e1e/pkg/purl/purl.go#L211

Reply:
Oh, yes, that's correct. Thanks.